marketplace_builder/views/pages/tutorials/custom-model-types/using-geojson-format-geolocation.liquid

```diff
@@ -0,0 +1,204 @@
+
+```graphql
+mutation createCompany($location: JSONPayload, $city: String, $timezone: String, $name: String){
+  customization_create(
+    customization: {
+      custom_model_type_name: "company"
+      properties: [
+        { name: "name" value: $name }
+        { name: "city" value: $city }
+        { name: "timezone" value: $timezone }
+        { name: "location" value_json: $location }
+      ]
+    }
+  ) {
+    id
+  }
+}
+```
+
+Create a `payload`:
+
+```json
+{
+  "name": "Company Name",
+    "city": "Boston",
+    "timezone": "Timezone",
+    "location": {
+      "type": "Point",
+      "coordinates": [-111.89903, 33.50921]
+  }
+}
+```
+
+### Step 3: Prepare GraphQL query
+
+Create the GraphQL query `graph_queries/search.graphql`:
+
+```graphql
+query search($point: GeometryJSON, $distance: Int) {
+  companies: models(
+    per_page: 100
+    filter: {
+      properties: [
+        { name: "location", distance_sphere: {center: $point, distance_in_km: $distance}}
+      ]
+    }) {
+    results {
+      properties
+    }
+  }
+}
+```
+
+Alternatively, you can create the record using a Form Configuration:
+
+```yaml
+---
+name: create_company
+fields:
+  properties:
+    name:
+      validation: { presence: true }
+    city:
+      validation: { presence: true }
+    timezone:
+      validation: { presence: true }
+    location:
+      validation: { geometry: true }
+```
+
+The HTML field snippet for geometry JSON is just a textarea:
+
+```html
+<div class="form-group">
+  <label for="location">Location</label>
+  <textarea required="true" id="location" type="text" class="form-control" name="{{ form_builder.fields.properties.location.name }}">
+      {{- context.params.form.properties_attributes.location }}
+  </textarea>
+</div> 
+```
+
+{% capture content %}
+This example is for presentation only. In a real scenario, no one would provide a raw JSON as a location address — there would be a JavaScript function using external geolocation services for getting coordinates and then building the required GeometryJSON structure.
+{% endcapture %}
+{% include 'alert/note', content: content %}
+
+### Step 4: Create page
+
+To wrap it up, create a page `views/pages/search.liquid`:
+
+```liquid
+{% raw %}
+---
+slug: search
+---
+{% parse_json location %}
+{
+  "type": "Point",
+  "coordinates": [
+    -111.89903,
+    33.50921
+  ]
+}
+{% endparse_json %}
+{% assign distance = context.params.distance | default: 100 | plus: 0 %}
+
+{% graphql g = 'search', point: location, distance: distance %}
+
+{% for company in g.companies.results %}
+[...]
+{% endfor %}
+{% endraw %}
+```
+
+## Live example and source code
+
+**Examples**
+
+* Form: [https://alpha.staging.oregon.platform-os.com/companies/new](https://alpha.staging.oregon.platform-os.com/companies/new)
+* Working advanced example: [https://alpha.staging.oregon.platform-os.com/companies/search](https://alpha.staging.oregon.platform-os.com/companies/search)
```

Comment:
Those links are to Darek's dev env. We should link to `examples.platform-os.com`